### ISSUE TITLE
Federated GraphQL failes with ObjectWithMetadata error

### ISSUE THREAD

**Issue by benoj** [mirumee/saleor#6921]
### What I'm trying to achieve
Trying to use Apollo Federated Gateway with saleor/

### Steps to reproduce the problem
1. create the following `docker-compose.yml` file
```
services:
  gateway:
    image: mirumee/gql-federation-gateway
    ports:
      - 4000:4000
    environment:
      - ENDPOINTS=first:https://demo.saleor.io/graphql/
```
2. docker-compose up

### What I expected to happen
Federated graphql workls

### What actually happens
```
Encountered error when loading first at https://demo.saleor.io/graphql/: Syntax Error: Unexpected Name "ObjectWithMetadata"

```

**System information**
docker

**Comment by fowczarek** [comment]
Resolved #7167

**Comment by fowczarek** [comment]
Resolved #7167
